build_tools/setup_helpers/extension.py

```diff
@@ -43,7 +43,10 @@ def _get_build(var, default=False):
 
 
 def get_ext_modules():
-    return [Extension(name='torchaudio._torchaudio', sources=[])]
+    return [
+        Extension(name='torchaudio.libtorchaudio', sources=[]),
```

Comment:
are there situations in which only the `libtorchaudio` C++ extension will be needed instead of both? I'm trying to understand if there's a reason why we're splitting the extension into separate C++ and Python components or if this is simply a refactor.

Reply:
> are there situations in which only the libtorchaudio C++ extension will be needed instead of both?

In C++ examples and mobile deployment, where they do not want to bring the whole Python library.

>  I'm trying to understand if there's a reason why we're splitting the extension into separate C++ and Python components or if this is simply a refactor.

It's somewhat of both but aiming for distributing `libtorchaudio` as a part of our Python package binary, so that integration with C++ code become easier, similar to PyTorch core does. Right now, when one wants to run torchaudio's custom implementation in C++ via TorchScript, they have to build `libtorchaudio` from scratch. By splitting the library and extracting the Python portion to `_torchaudio`, one can simply reuse the `libtorchaudio`.
Ideally, we should also provide `findTorchAudio.cmake` which is analogous to PyTorch core's `torch.utils.cmake_prefix_path`, to provide the same experience as PyTorch core, but that's a future work.

Reply:
from this PR implementation, it looks like `libtorchaudio` can not be built as a standalone but only by default as part of the `_torchaudio` Python extension (see [here](https://github.com/pytorch/audio/pull/1752/files#diff-dc5272c8001918aecaca882d26285022fa5179d9fd7c2268805a263525269179R63)), so for the first point (C++/mobile), is it possible to only bring in the C++ extension without the Python extension?

Reply:
`libtorchaudio` is always built. Check out the resulting [CMakeLists.txt](https://github.com/mthrok/audio/blob/split-lib/torchaudio/csrc/CMakeLists.txt) and notice that the libtorchaudio portion is not guarded by any if clause. On contrary, `_torchaudio.so` is built only when `BUILD_TORCHAUDIO_PYTHON_EXTENSION=1`, which is set by `setup.py`.